tests/operator_test.go

```diff
@@ -2081,7 +2083,22 @@ spec:
 				secret, err := coreClient.Secrets("openshift-monitoring").Get(context.Background(), "prometheus-k8s", metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
 
-				data := secret.Data["prometheus.yaml"]
+				data, ok := secret.Data["prometheus.yaml"]
+				// In new versions of prometheus-operator, the configuration file is compressed in the secret
```

Comment:
@erkanerol Is there a chance this test might be run against new or old prometheus operator code depending on env? Do we need to be smart and auto-detect which case to use?

Reply:
@stu-gott  I just checked it again and tried to find a smarter way but I couldn't find. Do we know which prometheus-operator versions we support? 

By the way, as far as I understand, we are not running this test in any upstream precommit job. See https://search.ci.kubevirt.io/?search=2936&maxAge=336h&context=1&type=all&name=&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

@dbasunag catched it in OpenShift Virtualization pipelines. 

We plan to add some specific lanes for metrics and alerts. I guess we will have better results in that case.

Reply:
Apologies. I misread. This code already does exactly what I asked. if data["prometheus.yaml"] exists, it's used. otherwise it looks for the gzip version. so it already works with both old and new versions of prometheus operator.